app/kubernetes/react/views/networks/ingresses/queries.ts

```diff
@@ -0,0 +1,162 @@
+import { useQuery, useMutation, useQueryClient } from 'react-query';
+
+import { EnvironmentId } from '@/portainer/environments/types';
+import {
+  mutationOptions,
+  withError,
+  withInvalidate,
+} from '@/react-tools/react-query';
+
+import {
+  getIngresses,
+  getIngress,
+  createIngress,
+  deleteIngresses,
+  updateIngress,
+  getIngressControllers,
+} from './service';
+import { DeleteIngressesRequest, Ingress } from './types';
+
+const ingressKeys = {
+  all: ['environments', 'kubernetes', 'namespace', 'ingress'] as const,
+  namespace: (
+    environmentId: EnvironmentId,
+    namespace: string,
+    ingress: string
+  ) => [...ingressKeys.all, String(environmentId), namespace, ingress] as const,
+};
+
+export function useIngress(
+  environmentId: EnvironmentId,
+  namespace: string,
+  name: string
+) {
+  return useQuery(
+    [
+      'environments',
+      environmentId,
+      'kubernetes',
+      'namespace',
+      namespace,
+      'ingress',
+      name,
+    ],
+    async () => {
+      const ing = await getIngress(environmentId, namespace, name);
+      return ing;
+    },
+    {
+      ...withError('Unable to get ingress'),
+    }
+  );
+}
+
+export function useIngresses(
+  environmentId: EnvironmentId,
+  namespaces: string[]
+) {
+  return useQuery(
+    [
+      'environments',
+      environmentId,
+      'kubernetes',
+      'namespace',
+      namespaces,
+      'ingress',
+    ],
+    async () => {
+      if (!namespaces.length) {
+        return [];
+      }
```

Comment:
could this be replaced with the 'enabled' option?
https://tanstack.com/query/v4/docs/guides/dependent-queries

Reply:
Yes. Updated.